### ISSUE TITLE
fix: wcag errors in Menu/MenuButton when using a FluentMenuProvider

### ISSUE THREAD

**Issue by radmorecameron** [microsoft/fluentui-blazor#3813]
# 🐛 Bug Report

Menus/MenuButtons are not accessible with just a keyboard when using a FluentMenuProvider 

## 💻 Repro or Code Sample

- https://www.fluentui-blazor.net/Menu#serviceprovider : tab to the example not using the menu provider and try to get to one of the menu items and select it using keyboard navigation only (**Tab**, **Shift+Tab**, **Arrow Keys**, **Space/Enter** key)
- https://www.fluentui-blazor.net/MenuButton#menubuttontoselectac : tab to the example not using the menu provider and try to get to one of the menu items and select it using keyboard navigation only (**Tab**, **Shift+Tab**, **Arrow Keys**, **Space/Enter** key)

## 🤔 Expected Behavior

Interactions with the keyboard should be made possible, correct aria-roles should be added so screenreaders can work better as well.

https://www.w3.org/WAI/ARIA/apg/patterns/menubar/ for menu
https://www.w3.org/WAI/ARIA/apg/patterns/menubar/#wai-ariaroles,states,andproperties for aria-roles that should be used on menu

https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/#kbd_label for menu button
https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/#rps_label for aria-roles that should be used on menu button

## 😯 Current Behavior

While using the MenuProvider, the tab order gets pretty messed up and there doesn't seem to be code to handle it properly

## 💁 Possible Solution

Follow the spec: 
https://www.w3.org/WAI/ARIA/apg/patterns/menubar/ for menu
https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/#kbd_label for menu button

Handle focusing the correct element and when focusing away, head to the next focusable element after the menu instead of where the menu provider puts the menu, etc.

## 🔦 Context

It can't be used by people who are visually impaired or that use alternative ways to navigate the web that don't involve a mouse (ex: screen reader)

## 🌍 Your Environment

Windows 11
Microsoft Edge
.NET 8
Fluent UI version: 4.11.9

**Comment by dvoituron** [comment]
I don't think we'll be able to fix this in version 4. These menus are created using Web Components, and we don't have the ability to customize everything.
Using **FluentMenuProvider** puts the HTML code at the end of the page, which makes it difficult to use Accessibility correctly.

You can choose not to enable **FluentMenuProvider** (via `UseMenuService=“false”`) or wait for the next version (v5), which handles this correctly. https://fluentui-blazor-v5.azurewebsites.net/Menu

**Comment by radmorecameron** [comment]
Sounds good, I didn't see any issues created that were related to this so I wasn't sure if this was a known issue. Since it is already fixed but not released yet, I'll close this issue. Thanks for the quick reply :)
